webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx

```diff
@@ -168,15 +169,26 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
     useApiRequestCompletionState({
         serviceName: Constants.pluginApiServiceConfigs.checkSubscriptionsConfigured.apiServiceName,
         handleError: (error) => {
-            dispatch(resetGlobalModalState());
-            if (
-            error?.id !== Constants.ApiErrorIdSubscriptionsNotConfigured &&
-            error?.id !== Constants.ApiErrorIdSubscriptionsUnauthorized &&
-            error?.id !== Constants.ApiErrorIdNotConnected
-            ) {
-                return handleErrorComponent(error);
+            if (error) {
+                if (error?.id === Constants.ApiErrorIdRefreshTokenExpired) {
+                    dispatch(setConnected(false));
+                    setShowErrorComponent(true);
+                    setApiError(error);
+                    return;
+                }
+
+                if (
+                    error?.id !== Constants.ApiErrorIdSubscriptionsNotConfigured &&
+                    error?.id !== Constants.ApiErrorIdSubscriptionsUnauthorized &&
+                    error?.id !== Constants.ApiErrorIdNotConnected
+                ) {
```

Comment:
```suggestion
```

Reply:
i don't think we can remove these conditions as we are not showing modal in these cases